sdk/go/common/workspace/templates_zip.go

```diff
@@ -87,6 +88,17 @@ func RetrieveZIPTemplateFolder(templateURL *url.URL, tempDir string) (string, er
 	if err != nil {
 		return "", err
 	}
+
+	var pulProjFile *zip.File
+	for _, f := range archive.File {
+		if !f.FileInfo().IsDir() && f.FileInfo().Name() == "Pulumi.yaml" {
+			pulProjFile = f
+		}
+	}
+	if pulProjFile == nil {
+		return "", errors.New("template does not have Pulumi.yaml file")
```

Comment:
This is consistent with that `new` errors for file paths if they don't have Pulumi.yaml so seems reasonable.